src/Symfony/Component/JsonEncoder/DataModel/Decode/ScalarNode.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonEncoder\DataModel\Decode;
+
+use Symfony\Component\TypeInfo\Type\BackedEnumType;
+use Symfony\Component\TypeInfo\Type\BuiltinType;
+
+/**
+ * Represents a scalar in the data model graph representation.
+ *
+ * Scalars are the leaves of the data model tree.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+final readonly class ScalarNode implements DataModelNodeInterface
+{
+    public function __construct(
+        public BuiltinType|BackedEnumType $type,
```

Comment:
Accepting a BackedEnumType in a ScalarNode looks wrong to me. PHP enums are **not** scalars. Either enums should have their own data model node, or this class should be renamed (and description changed) because it does not represent a scalar.

Reply:
:+1:, I created a `BackedEnumNode`, things are a bit clearer that way, thanks!